nntrainer/layers/time_dist.cpp

```diff
@@ -23,6 +23,63 @@ namespace nntrainer {
 
 const std::string TimeDistLayer::type = "time_dist";
 
+static void reshape(Tensor &m) {
+  TensorDim d = m.getDim();
+  m.reshape({d[2], d[1], d[0], d[3]});
+}
+
+void TimeDistLayer::setPosition() {
+  positions[0] = net_input[0]->getVariableRef().getData();
+  positions[1] = net_input[0]->getGradientRef().getData();
+  positions[2] = net_hidden[0]->getVariableRef().getData();
+  positions[3] = net_hidden[0]->getGradientRef().getData();
+}
+
+void TimeDistLayer::transposeInOut() {
+
+  Tensor &input_ = net_input[0]->getVariableRef();
+  input_.copy(transposeTensor(input_));
+
+  Tensor &ret_ = net_input[0]->getGradientRef();
+  if (ret_.getData() != positions[0]) {
+    ret_.copy(transposeTensor(ret_));
+  } else {
+    reshape(ret_);
+  }
+
+  Tensor &hval_ = net_hidden[0]->getVariableRef();
+  if (hval_.getData() != positions[0] && hval_.getData() != positions[1]) {
+    hval_.copy(transposeTensor(hval_));
+  } else {
+    reshape(hval_);
+  }
+
+  bool trans = true;
+
+  Tensor &derivative_ = net_hidden[0]->getGradientRef();
+  for (unsigned int i = 0; i < 3; ++i) {
+    if (derivative_.getData() == positions[i]) {
+      trans = false;
+      break;
+    }
+  }
+  if (trans)
+    derivative_.copy(transposeTensor(derivative_));
+  else
+    reshape(derivative_);
+}
+
+Tensor TimeDistLayer::transposeTensor(Tensor &m) {
+  TensorDim dim = m.getDim();
+
+  m.reshape({dim[1], dim[0], dim[2], dim[3]});
+  Tensor in = m.transpose("1:0:2");
+  in.reshape({dim[2], dim[1], dim[0], dim[3]});
```

Comment:
I wonder this is to transpose batch axis, if it is, it is okay if we don't change the underlying data but just reshape?

Reply:
I guess this relies on the assumption that channel stays 1 from the usage. Could you comment this somewhere in the function?

Reply:
right. done.